src/js/lib/content/src/templates.ts

```diff
@@ -0,0 +1,245 @@
+    const templatesView = require('./templates-view.html')
+    hideRootHtml()
+    document.body.innerHTML += templatesView
+    statusBox = document.getElementById('templates-status-box')
+
+    try {
+        await loadProjectsList(openProject)
+    } catch (error) {
+        displayStatusBox("Failed to load projects.")
+    }
+
+    setTemplateCardHandlers(openProject)
+}
+
+/**
+ * Remove the top-level div elements from the scene.
+ */
+function hideRootHtml(): void {
+    const matches = document.body.querySelectorAll('div')
+    matches.forEach(element => {
+        hiddenElements.push(element)
+        element.remove()
+    })
+}
+
+/**
+ * Restore the elements removed by the `hideRootHtml` function.
+ */
+function restoreRootHtml(): void {
+    let templatesView = document.getElementById('templates-view')
+    hiddenElements
+        .slice()
+        .reverse()
+        .forEach(element => document.body.prepend(element))
+    hiddenElements = []
+    templatesView.remove()
+}
+
+/**
+ * Show the message in the statsus box div element.
+ *
+ * @param text the message to display
+ */
+function displayStatusBox(text: string): void {
+    statusBox.innerHTML = text
+    statusBox.style.visibility = 'visible'
+}
+
+/**
+ * Clear the status box div element.
+ */
+function clearStatusBox(): void {
+    statusBox.style.visibility = 'hidden'
+}
+
+/**
+ * Load the projects list.
+ *
+ * Uses Project Manager to get the list of user projects and displays
+ * them in the projects side menu.
+ *
+ * @param openProject the callback that opens IDE with the provided project
+ */
+async function loadProjectsList(openProject: (project: string) => void): Promise<void> {
+    const projectsListNode = document.getElementById(PROJECTS_LIST)
+
+    const newProjectNode = document.getElementById(PROJECTS_LIST_NEW_PROJECT)
+    newProjectNode.setAttribute('style', 'cursor: pointer;')
+    newProjectNode.onclick = () => {
+        clearStatusBox()
+        PM.createProject('Unnamed', 'default')
+            .then((response: any) => {
+                if (response.error !== undefined) {
+                    console.error('Project manager openProject failed', response)
+                    displayStatusBox(response.error.message)
+                } else {
+                    restoreRootHtml()
+                    openProject(response.result.projectName)
+                }
+            })
+            .catch((error: any) => {
+                console.error('onclick', PROJECTS_LIST_NEW_PROJECT, error)
+                displayStatusBox("Failed to create a new project.")
+            })
+    }
+
+    const projectsListResult = await PM.listProjects()
+    const projectsList = projectsListResult
+        .result
+        .projects
+        .map((project: any) => buildProjectListNode(project.name, openProject))
+
+    projectsList.forEach((element: any) => {
+        projectsListNode.insertBefore(element, newProjectNode)
+    })
+}
```

Comment:
Project listing / opening functionality is already implemented in Rust. We've been talking about implementing only HTML in view and all logic in Rust – we've been talking in what files to implement it and I remember describing how to generate HTML elements from rust and then how to connect callbacks to it. We were never talking about removing the logic and moving it to JS. This code here reimplements things that are in rust and introduces new errors. I never had a design meeting when we were changing Rust to JS. So this should go back to Rust (until we have a design meeting and discuss that otherwise). Anyway - reimplementing the same logic twice in two languages is error-prone and it should not be done this way.

Reply:
It's probably not possible to move it back to Rust tomorrow (while we plan to have the release). See my additional notes as a separate comment to this PR.